### PR TITLE
[Model] Support SenseNova U1 (NEOChatModel, MoT interleaved understanding+generation)

### PR DESCRIPTION
**Pull request by waltstephen**
## Summary

This PR adds an HF-compatible SenseNova U1 integration to SGLang-Omni for `NEOChatModel` / MoT interleaved understanding and generation.

The integration provides:

- `NEOChatModel` pipeline registration with the `SenseNovaU1` alias.
- A `u1_vqa` single-stage understanding pipeline using U1's native-resolution visual preprocessing and official HF-compatible math path.
- U1 hybrid image-span prefill attention helpers, including dense reference masking, image-span metadata, and M-block OR-reduction summaries.
- A `u1_flow` single-stage flow-matching image generation path for T2I and IT2I.
- A `u1_interleave` single-stage end-to-end interleaved text-image-text generation path.
- OpenAI-compatible image output plumbing: non-streaming chat returns `message.images`; streaming chat returns `delta.images`, with inline `image_url.url` data URLs.
- Unit coverage for SenseNova U1 pipeline registration, hybrid attention helpers, image metrics, and OpenAI-compatible image outputs.

The current implementation is intentionally HF-compatible and in-engine: it keeps SenseNova U1 inside the SGLang-Omni request/stage lifecycle while delegating U1 numerical kernels to the official `NEOChatModel` implementation. This establishes correctness and service/API behavior before a future native fused-attention / scheduler-performance pass.

## Why

SenseNova U1 is a unified MoT model for interleaved multimodal understanding and generation. Downstream rollout workloads need a single serving surface that can emit text and real generated image pixels in one response. The official deployment uses separate LightLLM + LightX2V engines; this PR instead exposes a single SGLang-Omni pipeline variant so interleaved generation can be used through SGLang-Omni's normal request and OpenAI-compatible serving paths.

## Implementation Notes

- `sglang_omni.models.sensenova_u1.config`
  - Registers `NEOChatModel` and alias `SenseNovaU1`.
  - Adds default, `flow`, and `interleave` variants.

- `sglang_omni.models.sensenova_u1.hf_runner`
  - Loads official SenseNova U1 model code from an installed package or `SENSENOVA_U1_VENDOR_ROOT`.
  - Uses `SENSENOVA_U1_MODEL_PATH` or the public HF repo id by default.
  - Patches compatibility gaps between the public U1 code and newer Transformers APIs.
  - Builds U1 native-resolution VQA prompts with `/<IMG_CONTEXT></img>` expansion.

- `sglang_omni.models.sensenova_u1.hybrid_attention`
  - Implements the U1 dense hybrid mask: text rows stay causal; image rows can attend to the complete image span sharing the same temporal index.
  - Provides a dense attention reference used in parity tests.

- `sglang_omni.models.sensenova_u1.flow_matching`
  - Wraps U1 T2I/IT2I flow-matching generation in a SGLang-Omni stage.
  - Exposes scheduler parameters and image comparison helpers.

- `sglang_omni.models.sensenova_u1.interleave`
  - Wraps U1 `interleave_gen()` for text-image-text generation.
  - Returns `omni_rollout` metadata with generated images and text/image segments.

- OpenAI API plumbing
  - Carries `omni_rollout` through client stream chunks.
  - Adds `image_config`, `chat_template_kwargs`, and `n` fields to chat requests.
  - Emits generated images as OpenAI-style `image_url` objects in non-streaming and streaming chat responses.

## Validation

All evidence was produced on a single local H100 80GB with project-local environments only.

### M1 Understanding

- Weight loading: zero missing, unexpected, mismatched, or error keys.
- MoT generation parameters loaded: `551`.
- Logits parity vs official HF path: cosine `0.999992847442627`, max abs diff `0.0`.
- Greedy VQA decode: 20/20 prompts token-exact vs HF.
- Throughput: SGLang-Omni fallback `17.954721 tokens/s`; HF reference `18.056931 tokens/s`.

### M2 Hybrid Attention

- Dense mask unit cases passed.
- Hybrid attention parity vs official HF eager attention:
  - mask max abs diff `0.0`
  - attention output max abs diff `0.0`
  - attention weights max abs diff `0.0`
- Interleaved two-image greedy prompts: 3/3 token-exact.

### M3 Flow Matching

- T2I image parity vs HF:
  - pixel max abs diff `0`
  - PSNR `Infinity`
  - SSIM `1.0`
- IT2I image parity vs HF:
  - pixel max abs diff `0`
  - PSNR `Infinity`
  - SSIM `1.0`
- Scheduler parameters were logged and matched the U1 config, including `timestep_shift=1.0`, `time_schedule="standard"`, `time_shift_type="exponential"`, `base_shift=0.5`, `max_shift=1.15`, `noise_scale_mode="resolution"`, `P_mean=-0.8`, `P_std=0.8`, and `t_eps=0.05`.

### M4 Interleaved Generation

- 10/10 interleaved prompts passed.
- Text exact match vs HF: 10/10.
- Token exact match vs HF: 10/10.
- Each prompt generated one real PNG image.
- Image parity for every generated image:
  - pixel max abs diff `0`
  - PSNR `Infinity`
  - SSIM `1.0`
- OpenAI-style service probe returned HTTP 200 and emitted `delta.images[0].image_url.url`.
- Sequential batch throughput:
  - `0.0522067608 images/s`
  - `10.2377457853 tokens/s`
  - peak CUDA allocated/reserved `33.9568457603 / 34.5703125 GiB`

### M5 Unit / Regression Tests

Focused SenseNova U1 and image-output tests:

```bash
python -m pytest \
  tests/unit_test/sensenova_u1 \
  tests/unit_test/serve/test_openai_api.py::test_chat_request_passes_image_generation_controls \
  tests/unit_test/serve/test_openai_api.py::test_chat_completion_non_stream_returns_inline_images \
  tests/unit_test/serve/test_openai_api.py::test_chat_completion_stream_returns_inline_image_delta \
  -q
```

Result: `9 passed, 1 warning`.

OpenAI API regression file:

```bash
python -m pytest tests/unit_test/serve/test_openai_api.py -q
```

Result after providing FFmpeg shared libraries for `torchcodec`: `94 passed, 1 warning`.

## Known Limitations

- This is an HF-compatible in-engine fallback. It validates model behavior, image pixels, interleaved generation semantics, and OpenAI-compatible service shape, but it does not yet claim a native fused FlashInfer/RadixAttention U1 kernel.
- The hybrid attention implementation in this PR is a dense Python reference/helper. A production fused backend for the U1 image-span row policy remains future work.
- Flow-matching generation runs inside a single terminal stage with `max_concurrency=1` in the validated configuration. It does not yet implement cooperative per-flow-step interleaving with the native autoregressive scheduler.
- Throughput is correctness-oriented rather than optimized; the measured M4 batch wall throughput was `0.0522067608 images/s`.

### CODE REVIEW

**Comment by waltstephen** [comment]
Converting this to draft. Flagging a correctness issue with my own PR before reviewers spend time on it: the current branch registers a SenseNova U1 pipeline in SGLang-Omni but delegates the actual model math to the HuggingFace `NEOChatModel` implementation rather than running natively on the SGLang engine.

Concretely, this means the branch as it stands does NOT deliver what the title claims:
- No native SGLang model definition (no `forward(..., forward_batch)` over SGLang parallel linear layers / RadixAttention / paged KV).
- The hybrid image-span attention mask is a standalone dense Python reference, not wired into a fused attention backend on the serving path.
- No continuous batching (`max_concurrency=1`).
- Measured throughput is 17.95 tok/s vs 18.06 tok/s for plain HF transformers — i.e. no speedup at all.
- The reported numerical "parity" (max abs diff 0.0, PSNR inf, SSIM 1.0, identical PNG hashes) is an artifact of both sides executing the same HF code path, so it does not validate a port.

I am reworking this into a real native SGLang implementation and will only mark it ready for review once it shows a genuine speedup over HF transformers. Apologies for the noise.
